### ISSUE TITLE
(CLI) `typst watch` no longer works

### ISSUE THREAD

**Issue by jwhear** [typst/typst#1436]
### Description

The `watch` command has been one of my favorite typst features for several months now.  But in the last few days it has stopped working: running it on a document causes an initial compile but modifications to the file no longer cause recompilation or any update in the typst output.

I build typst from source and am currently on tag `release-0.4.0`; this was broken before but I'm not sure what commit I was previously using.  I'm on up-to-date Fedora 37, kernel 6.3.5.
```
$ typst --version
typst 0.4.0 (488ded0f)
```

Running `typst -vvv watch document.typ` produces one line of trace:
```
TRACE mio::poll: registering event source with poller: token=Token(0), interests=READABLE
```
Modifying and saving the document doesn't cause any additional logging to appear or any output change at all.

I don't know what `mio::poll` is using under the hood, but running 
```
inotifywait -e MODIFY document.typ
```
works as expected, exiting when I save the file.

Finally, this behavior is consistent regardless of file or path; none of the paths I'm using are symlinks, so it's not #301 .

I'm a software developer and comfortable with Rust, so happy to go deep in diagnosing and fixing this issue if there's a good starting point.

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by jwhear** [comment]
Built a debug version and ran it under rust-gdb.  It blocks waiting for events here:
```
#13 0x00005555557ba9a9 in typst::compile (command=...)
    at cli/src/main.rs:208
```

I told GDB to step and tried modifying the watched file; the step did not finish, so apparently events are not showing up in the channel.  I'm not really familiar with the typst codebase, but this seems to point strongly to an issue in the notify dependency.

**Comment by LaurenzV** [comment]
That's weird, and in the past it has worked fine for you? If so, I think maybe one way of identifying the issue would be to go back to a commit where it worked fine for you and then work your way up until you can find the commit which breaks the `watch` command for you.

**Comment by jwhear** [comment]
I tracked it down to commit 8300f75f22cb4dd1b871a7e617397ff7d56ba001 (earliest commit when watch stopped working), read the commit message, and had an aha moment.  I recently set `TYPST_ROOT` to a directory where I keep all my templates; unsetting the variable causes watch to begin working again.  I can quickly repro by using the `--root` argument when invoking typst.

Am I misusing the root feature or is this unintended?  Perhaps typst is watching the root directory rather than the current working directory?
